Fix #50044 - Updates dashcard loading state so it shows the correct title while it's loading

bpander opened the pull request:
Closes https://github.com/metabase/metabase/issues/50044
Closes UXW-105

### Description

Updates dashcard title fallback so it shows the correct title while it's loading.

### How to verify

#50044 has good repro steps but you can also reproduce the issue if you throttle your network.

You can also reproduce the issue by changing the title via "Visualize another way".

### Demo

**BEFORE**

https://github.com/user-attachments/assets/51677df1-f0a1-4fa2-b298-88cf7a1b26b9

**AFTER**

https://github.com/user-attachments/assets/ae8813d0-9a35-4712-8ebb-2eac844be48f

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

bpander committed:
UXW-105 Updates dashcard loading state so it shows the correct title while it's loading

npfitz reviewed (approved):
I'm fine with this as it is, but from my little bit of testing, I couldn't get `settings` to have anything in it at all, so maybe it should just be removed? Also, could we have a unit test in `ChartCaption.unit.spec` to cover the different sources of card names?

bpander committed:
UXW-105 Adds unit tests

bpander commented:
@npfitz 
> I couldn't get `settings` to have anything in it at all, so maybe it should just be removed?

The one in ChartCaption? That always has stuff in it for me (once the card data is loaded). That's why the card titles were correct when the card finished loading. I'd rather not mess with that. This bug only deals with the loading state; if the fallback I added is bad, it's at least only wrong while the card is loading. If we mess with `settings`, we might break the loaded state too.

<img width="1227" height="946" alt="Screenshot 2025-11-03 at 2 33 03 PM" src="https://github.com/user-attachments/assets/39b112bc-4955-408e-bf9e-8b7ff0c6fe9d" />


---


> could we have a unit test in `ChartCaption.unit.spec` to cover the different sources of card names?

Added in 3a191b2f33e.
